include/zephyr/drivers/gpio_keys.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GPIO_KEYS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GPIO_KEYS_H_
+
+#include <stdint.h>
+#include <zephyr/drivers/gpio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct gpio_keys_callback {
+	struct gpio_callback gpio_cb;
+	uint32_t zephyr_code;
+};
+
+typedef void (*gpio_keys_callback_handler_t)(const struct device *port,
+					     struct gpio_keys_callback *cb, gpio_port_pins_t pins);
+/**
+ * @brief GPIO Keys Driver APIs
+ * @defgroup gpio_keys_interface GPIO KeysDriver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+__subsystem struct gpio_keys_api {
+	int (*enable_interrupt)(const struct device *dev, gpio_keys_callback_handler_t cb);
+	int (*disable_interrupt)(const struct device *dev);
+	int (*get_pin)(const struct device *dev, uint32_t idx);
+	int (*get_pin_raw)(const struct device *dev, uint32_t idx);
```

Comment:
If this is going with a custom API, is there really the need for a `raw` function? May as well only use the logical one and simplify the code and structures a tiny bit.